modules/gdscript/editor/gdscript_docgen.cpp

```diff
@@ -543,7 +543,7 @@ void GDScriptDocGen::_generate_docs(GDScript *p_script, const GDP::ClassNode *p_
 
 				for (const GDP::EnumNode::Value &val : m_enum->values) {
 					DocData::ConstantDoc const_doc;
-					const_doc.name = val.identifier->name;
+					const_doc.name = String(name) + "." + String(val.identifier->name);
```

Comment:
This looks incorrect. A constant name (identifier) ​​can't contain a period. At the very least, this is a hack, not a desirable option.

The main problem with proper enumeration support is that the core doc system doesn't properly support scoped enumerations. Instead, all enumeration elements are put into a common constant list. Native enumeration elements cannot be repeated, but custom enumerations can have elements with the same name in multiple enumerations.